dockerfile_scripts/install_libnss_determined.sh

```diff
@@ -5,5 +5,19 @@ set -e
 # Add a plugin to the user system that lets us extend the users available in
 # the container at runtime. This is critical for supporting non-root shell,
 # which in turn is critical for non-root distributed training.
+#
+# Also, to work around a behavior in OpenShift 4.+ [1], place `determined`
+# NSS plug-in to be searched first instead of last so Determined-defined
+# entries are used when there are conflicts with infrastructure-based
+# entries in these identity databases: passwd, shadow, group.
+#
+# [1] https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids "By default,
+# OpenShift 4.x appends the effective UID into /etc/passwd of the Container
+# during the creation of the Pod."
+
 make -C /tmp/det_dockerfile_scripts/libnss_determined libnss_determined.so.2 install \
-    && sed -E -i -e 's/^((passwd|shadow|group):.*)/\1 determined/' /etc/nsswitch.conf
+    && sed -i -E '
+        /^(passwd|group|shadow):/{
+            s/[ \t]determined//;/^(passwd|group|shadow):/;
+            s/:([ \t]+)/:\1determined /;
+        }' /etc/nsswitch.conf
```

Comment:
this doesn't look quite right... there seem to be two copies of `/^(passwd|group|shadow):/;` and I don't think they're placed in a particularly meaningful way.

Reply:
Yeah, there's a selector with no action there - which I'm pretty sure is not the intention. :)

Reply:
(groan) Editing error.  Updated to

```
        /^(passwd|group|shadow):/{
            s/[ \t]determined//;
            s/:([ \t]+)/:\1determined /;
        }' /etc/nsswitch.conf
```

Reply:
I'm clearly partial to my suggestion with the colon inside the capture group instead of repeating the colon in the replacement.  But either way, this looks better.

Reply:
Done.

```
        /^(passwd|group|shadow):/{
            s/[ \t]determined//;
            s/(:[ \t]+)/\1determined /;
        }' /etc/nsswitch.conf
```